src/ssh-generator/ssh-generator.c

```diff
@@ -103,14 +106,19 @@ static int make_sshd_template_unit(
                 if (r < 0)
                         return r;
 
+                /* only change AuthorizedKeysFile if an ephemeral key is present */
+                if (access("/run/ssh.authorized_keys.root", F_OK) >= 0)
+                        authorized_keys_file = " -o 'AuthorizedKeysFile /run/ssh.authorized_keys.root .ssh/authorized_keys'";
+
```

Comment:
so, i don't think tmpfiles.d/ sould be involved in any of this. Instead, just do ImportCredential to import the credential directly into the sshd instance. And then use `${CREDENTIALS_DIRECTORY}/…` to reference it directly from there in the sshd command line.

And I think it would make sense to just extend the command line like that unconditionally. as i understand AuthorizedKeysFile on a non-existing file is handled gracefully by sshd, hence we can just specify this basially always

Reply:
Okay thats perfect, I wasn't aware you could do ${CREDENTIALS_DIRECTORY}. the way I wanted to implement this originally was by referring to the credential file itself so I'm happy its actually able to be implemented like that!